scripts/release.sh

```diff
@@ -63,6 +63,11 @@ main() {
   log_callout "BRANCH=${BRANCH}"
   log_callout "REPOSITORY=${REPOSITORY}"
   log_callout ""
+
+  if [ "${BRANCH}" != "main" ] && [ "${BRANCH}" != "release-3.5" ]; then
```

Comment:
Just curious how do you or someone else release 3.4.x?

Reply:
Goal is to give a clear signal that this script was not tested and is not expected to work with `release-3.4`. Instead of failing in random part of the execution for v3.4, goal is to give release manager signal that this script needs to be fixed.

Reply:
Do you assume in general usage of script from 'main' against 3.5 branch ? Why this is a goal ?

One of the principles it breaks is 'build determinism'. If someone sync's to given sha on given branch and perform release, we expect the the produced artifacts to be as much identical as possible (i.e. golang version for building binaries is enforced), vs. pick part of the release logic from the 'head' branch.